src/expr/node_traversal.h

```diff
@@ -58,9 +58,9 @@ class NodeDfsIterator
   // Dereference
   reference operator*();
   // Equals
-  bool operator==(const NodeDfsIterator&) const;
+  bool operator==(const NodeDfsIterator&);
```

Comment:
feels a bit confusing. Is there a clean way to keep these operators `const`? If not, perhaps this is worth documenting in a comment? Or am I missing something standard?

Reply:
Unfortunately, I believe that this is required, and is the cost of have `begin()` be constant-time.
I'll write about it.

Reply:
Thanks for this comment! While writing the docs, I realized that I forgot to initialize `other`.

Reply:
Per an offline discussion with Yoni, we'd like you to take a look at this, @mpreiner. Do you think there might be some way to keep the `operator==` implementation `const`, with allowing for uninitialized iterators?

I can imagine one or two, but they're pretty bad.

Reply:
Okay, Mathias and I talked about this.

He thinks that this weirdness should be documented, but may be unavoidable.